### PR TITLE
feat(text-field): add component and demo

### PR DESCRIPTION
**Pull request by Flo5k5**

### CODE REVIEW

File: src/components/text-field/react/TextField.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                        // tslint:disable-next-line: jsx-no-lambda
+                        onBlur={(): void => setHasFocus(false)}
+                        onChange={handleChange}
+                        {...props}
+                    />
+                </div>
+
+                {(isValid || hasError) && (
+                    <Icon
+                        className={'__input-validity"'}
```
Comment (mcastagnetti) [inline src/components/text-field/react/TextField.tsx]: Seems wrong classname

File: src/components/text-field/react/TextField.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+    return (
+        <div
+            className={classNames(className, handleBasicClasses({ prefix: CLASSNAME, theme }), {
+                [`${CLASSNAME}--has-error`]: !isValid && hasError,
+                [`${CLASSNAME}--has-icon`]: icon,
+                [`${CLASSNAME}--has-label`]: label,
+                [`${CLASSNAME}--has-placeholder`]: placeholder,
+                [`${CLASSNAME}--has-value`]: hasValue,
+                [`${CLASSNAME}--is-disabled`]: isDisabled,
```
Comment (mcastagnetti) [inline src/components/text-field/react/TextField.tsx]: Can be handled by `handleBasicClasses` but don't know if its the right approach or not :man_shrugging:

File: src/components/text-field/react/TextField.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                [`${CLASSNAME}--is-focus`]: hasFocus,
+                [`${CLASSNAME}--is-valid`]: isValid,
```
Comment (mcastagnetti) [inline src/components/text-field/react/TextField.tsx]: idem `handleBasicClasses`

File: src/components/text-field/react/TextField.tsx (hunk 1 above)
Comment (Flo5k5) [reply src/components/text-field/react/TextField.tsx]: Nice catch

**Comment by matmkian** [comment]
Tests ok!

**Pull request merged**